test/python/WMCore_t/REST_t/Api_t.py

```diff
@@ -225,9 +225,23 @@ def test_multi_throw10(self):
 
 def setup_server():
     srcfile = __file__.split("/")[-1].split(".py")[0]
-    setup_test_server(srcfile, "Root")
-    #cpconfig.update({"log.screen": True})
+    server, T.test_authz_key = setup_test_server(srcfile, "Root")
+    print("SERVER", server.config.pythonise_())
+    print("AUTHZ", T.test_authz_key, T.test_authz_key.data)
+
+import cherrypy
+from cherrypy.test import webtest
```

Comment:
Eric,
I think something really screwed on bot environment. From this test:
https://cmssdt.cern.ch/jenkins/job/DMWM-PR-test/442/testReport/junit/(root)/Api_t/testmain/

I see the following:

/build/cmsbuild/jenkins-workarea/workspace/DMWM-PR-unittests/SLICE/1/label/slc6_amd64-cloud/deploy/1.0.9.pre9/sw.pre/slc6_amd64_gcc481/external/cherrypy/3.2.2-comp10/lib/python2.6/site-packages/cherrypy/process/servers.py",
line 187, in _start_http_thread     self.httpserver.start() AttributeError:
'str' object has no attribute 'start'

it means that someone overwrite self.httpserver which now is a string rather
then cherrypy object.

I built cherrypy in my VM, so I would assume I have similar environment and I
can run unit tests just fine. But on a bot it is different.

So, I want to see bot installation and environment. Do we have access to the
node that we can manually run the unittest from there within bot environment?


On  0, Eric Vaandering <notifications@github.com> wrote:
> > @@ -225,9 +225,23 @@ def test_multi_throw10(self):
> >  
> >  def setup_server():
> >      srcfile = __file__.split("/")[-1].split(".py")[0]
> > -    setup_test_server(srcfile, "Root")
> > -    #cpconfig.update({"log.screen": True})
> > +    server, T.test_authz_key = setup_test_server(srcfile, "Root")
> > +    print("SERVER", server.config.pythonise_())
> > +    print("AUTHZ", T.test_authz_key, T.test_authz_key.data)
> > +
> > +import cherrypy
> > +from cherrypy.test import webtest
> 
> Let's keep imports at the top unless this is just a test Valentin is trying to run (not really asking to merge)
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/dmwm/WMCore/pull/6183/files#r38317754